platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/BuildscriptResolutionIntegrationTest.groovy

```diff
@@ -32,9 +32,7 @@ class BuildscriptResolutionIntegrationTest extends AbstractIntegrationSpec {
         """
     }
 
-    // This is not desired behavior. A project dependency should refer to the actual
-    // project, not its buildscript component. This is a bug.
-    def "project buildscript configuration can select itself"() {
+    def "project buildscript configuration cannot select itself"() {
```

Comment:
🤔 Is the foo attribute necessary here?  If not, this and the test below could be combined into parameterized test.

Reply:
`foo` is necessary to cause the configuration to be treated as a variant, as configurations without attributes are not variants. The second test works since the `classpath` configuration already has attributes.